### ISSUE TITLE
Style/IndentationWidth reports weird indentation 

### ISSUE THREAD

**Issue by NekoNova** [bbatsov/rubocop#2765]
Hello,

I have the following snippet of code:

```
def cookies
    cookie_ids = if self.home_made? 
      Cookie.home_made_cookie_ids
    elsif self.store_bought?
      Cookie.store_bought_ids
    elsif self.grandma?
      Cookie.grandma_ids
    else
      raise NotImplementedError
    end

    Cookie::Base.where(id: cookie_ids)
  end
```

Which is how we currently code our assignments.
We have the following cops defined for alignment:

```
inherit_from: .rubocop_todo.yml
AllCops:
  Include:
    - Rakefile
    - config.ru
  Exclude:
    - db/schema.rb
  TargetRubyVersion: 2.3

Lint/EndAlignment:
  AlignWith: variable
```

this is reported in the console:
```
app/models/cookie.rb:8:7: C: Style/IndentationWidth: Use 2 (not -14) spaces for indentation.
      Cookie.home_made_cookie_ids
      ^^^^^^^^^^^^^^
```

I don't understand why this one is being triggered.
When I change the first line to indent 6 tabs, the Cop actually accepts the file, ignoring all other lines.

**Comment by alexdowad** [comment]
> I don't understand why this one is being triggered.

Because it expects that line to be indented relative to `if`, not relative to the first token in the previous line.

@jonas054 Do we have some kind of style to indent relative to the start of the previous line, rather than the `if` keyword? Do you think that is something we should add?

**Comment by NekoNova** [comment]
The thing is, this behaviour changed when I upgraded to the latest version, might have missed if this was a new cop. Adding a Cop for the indentation would be nice if this is an accepted standard in the Ruby community.

**Comment by jonas054** [comment]
@NekoNova The indentation style you use is the one that should be enforced by
```yaml
Lint/EndAlignment:
  AlignWith: variable
```
or, if you want
```yaml
Lint/EndAlignment:
  AlignWith: start_of_line
```
so I don't see why you got the offense report. Perhaps you could double check by making a minimal isolated example. Just a directory with the inspected file, `.rubocop_todo.yml`, and `.rubocop.yml`. If you still get the offense report, show us the full example.

**Comment by NekoNova** [comment]
yeah that's exactly what I was thinking as well, hence why I opened this issue, because the code was not being flagged in the previous version, but it is now.
I'll try to make an isolated issue on Monday with only that rule enabled and see if it still fails.

**Comment by NekoNova** [comment]
Issue was related to #1691
Switching to 
```
Lint/EndAlignment:
  AlignWith: variable
```

fixes it properly.

**Comment by brandonweiss** [comment]
I think I'm still running into this issue… I have:

```
Lint/EndAlignment:
  EnforcedStyleAlignWith: "start_of_line"
```

And if I turn on indentation width I get weird suggestions, like this:

```ruby
def carry_attribution_data_from_previous_customer_invoice
  items = if line_items.retrospective.pre_subtotal.any?
            line_items.retrospective.pre_subtotal
  else
    line_items.bill_ahead.pre_subtotal
  end
end
```

It's obviously trying to indent two lines relative to the `if` statement, but that's not what I'd want/expect. Is it colliding with some other rule?

**Comment by jonas054** [comment]
I'll try to give you the "teach a man to fish" answer.

If you run your example with default configuration and this command:

```
$ rubocop --auto-correct --display-cop-names /tmp/2765.rb 
```
you get
```
Inspecting 1 file
W

Offenses:

/tmp/2765.rb:2:3: W: Lint/UselessAssignment: Useless assignment to variable - items.
  items = if line_items.retrospective.pre_subtotal.any?
  ^^^^^
/tmp/2765.rb:4:3: C: [Corrected] Style/ElseAlignment: Align else with if.
  else
  ^^^^
/tmp/2765.rb:5:5: C: [Corrected] Style/IndentationWidth: Use 2 (not -6) spaces for indentation.
    line_items.bill_ahead.pre_subtotal
    ^^^^^^
/tmp/2765.rb:6:3: W: Lint/EndAlignment: end at 6, 2 is not aligned with if at 2, 10.
  end
  ^^^

1 file inspected, 4 offenses detected, 2 offenses corrected
```
and the code has been corrected to
```ruby
def carry_attribution_data_from_previous_customer_invoice
  items = if line_items.retrospective.pre_subtotal.any?
            line_items.retrospective.pre_subtotal
          else
            line_items.bill_ahead.pre_subtotal
  end
end
```
You'll note that the `end` was reported but not corrected. Why was that? You'll see it here:
```
$ rubocop --show-cops | grep -A8 Lint/EndAlignment
Lint/EndAlignment:
  Description: Align ends correctly.
  Enabled: true
  EnforcedStyleAlignWith: keyword
  SupportedStylesAlignWith:
  - keyword
  - variable
  - start_of_line
  AutoCorrect: false
```
The answer is that `AutoCorrect: false` is the default configuration for this cop. The reason behind it is described in a link from #3900.

**Comment by brandonweiss** [comment]
@jonas054 Thanks! But, I don't think I understand your answer… the outcome I was expecting was this:

```ruby
def carry_attribution_data_from_previous_customer_invoice
  items = if line_items.retrospective.pre_subtotal.any?
    line_items.retrospective.pre_subtotal
  else
    line_items.bill_ahead.pre_subtotal
  end
end
```

My setting for `Lint/EndAlignment` is:

```yaml
Lint/EndAlignment:
  EnforcedStyleAlignWith: "start_of_line"
```

But I don't understand how `Lint/EndAlignment` would be affecting the indentation of the first branch in the `if` statement. And if it was, given that the value is `start_of_line`, shouldn't it then be the outcome I was expecting, rather then indented two spaces relative to the `if` statement?

**Comment by jonas054** [comment]
Yes, you're right. I get the same result as you with `EnforcedStyleAlignWith: start_of_line`, while `EnforcedStyleAlignWith: variable` produces the result you want. This looks like a bug.

**Comment by brandonweiss** [comment]
@bbatsov Can you reopen this? Or should I make a new issue?

**Comment by bbatsov** [comment]
File a new ticket.
